### DISCUSSION TITLE
Is the java8-compatible version still supported/updated?

### DISCUSSION THREAD

**Discussion by Niko-sk2x** [falkreon/Jankson#68] [Q&A] [answered]
For now I still need the java 8 compatible version (because sadly my code that uses it is *still* stuck on Minecraft 1.12.2, despite working on updating for the last few years) and there are some issues I have with jankson there.

Is this version still maintained? If it is, then for how long?

**Comment by falkreon** [comment, marked as answer]
The `java8` branch is in extended support; bugfixes will target both branches whenever possible, but document access and other new features won't be coming to java 8.

The `main` branch targets the latest LTS java, which at the moment matches Minecraft at SE 17. The next LTS is expected to happen at the end of this year (September 2023), and I expect Mojang to adopt SE 21 fairly quickly after its release. At that point I will expect to have to drop a new branch for extended 17 support and I'll reevaluate what level of support I can give java 8.

Short answer: yes, `1.2.x` is and will continue to be safe to use for Minecraft 1.12.2. It will continue to get bugfixes as needed for at least another year.
